packages/rpc-types/src/commitment.ts

```diff
@@ -9,9 +11,9 @@ function getCommitmentScore(commitment: Commitment): number {
         case 'processed':
             return 0;
         default:
-            return ((_: never) => {
-                throw new Error(`Unrecognized commitment \`${commitment}\`.`);
-            })(commitment);
+            throw new SolanaError(SOLANA_ERROR__INVARIANT_VIOLATION_SWITCH_MUST_BE_EXHAUSTIVE, {
+                unexpectedValue: commitment satisfies never,
+            });
```

Comment:
This one smells a little bit. I guess the only way this gets thrown is if someone isn't using TypeScript (or ignores it) and sends a string that isn't a valid `Commitment`. Shouldn't the error say that, then? ie. `SOLANA_ERROR__INVALID_COMMITMENT_VALUE`?

Reply:
Yeah, I was trying to make something that was generally usable for exhaustive switches. The error location will tell the developer where the unhandled value was thrown, which I figured was enough paired with the actual unexpected value.

Reply:
https://dev.to/cefn/typescript-satisfies-never-exhaustiveness-checking-in-typescript-49-58fh

Reply:
I do actually like that we have utility errors for these sort of invariants.